src/Core/Search/Discussion/Gambit/AuthorGambit.php

```diff
@@ -50,7 +51,7 @@ protected function conditions(AbstractSearch $search, array $matches, $negate)
 
         $ids = [];
         foreach ($usernames as $username) {
-            $ids[] = $this->users->getIdForUsername($username);
+            $ids[] = is_numeric($username) ? $username : $this->users->getIdForUsername($username);
```

Comment:
Taking a look at this PR again it's about the authorId not a user id, let's keep it like that and stick to using an ID integer, not a username.